### PR TITLE
compiler: Tweak device-aware blocking

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: tests/test_dle.py [hunk 1]
```diff
@@ -216,6 +217,26 @@ def test_cache_blocking_structure_optrelax_customdim():
                      'd,x0_blk0,y0_blk0,z0_blk0,x,y,z')
 
 
+def test_cache_blocking_structure_optrelax_defaultdim_alone():
```
Comment (EdCaunt) [inline tests/test_dle.py:220]: This is a horrible test name. How about `test_blocking_defaultdim_blockrelax`? Failing that, give it a shorter name and add a docstring explaining what it does.

File: tests/test_dle.py [hunk 2]
```diff
@@ -216,6 +217,26 @@ def test_cache_blocking_structure_optrelax_customdim():
...
+
+    u = TimeFunction(name="u", grid=grid)
+    f = Function(name="f", grid=grid, dimensions=(d, x, y, z),
+                 shape=(2,) + grid.shape)
+
+    eqn = Inc(f, u*cos(time*d))
+
+    op = Operator(eqn, opt=('advanced', {'blockrelax': 'device-aware'}))
+
+    _, _ = assert_blocking(op, {'d0_blk0', 'x0_blk0'})
```
Comment (EdCaunt) [inline tests/test_dle.py:234]: Why do you need to create dummy variables here? Could you not just have `assert_blocking(op, {'d0_blk0', 'x0_blk0'})` on its own?

Comment (FabioLuporini) [reply tests/test_dle.py:234]: copy-paste leftover

File: tests/test_dle.py (hunk 1 above)
Comment (FabioLuporini) [reply tests/test_dle.py]: don't check the commit time because I had to rebase to address your other comment below. So you've to do with faith this time

**Review by EdCaunt — APPROVED** [review]

**Review by georgebisbas — APPROVED** [review]

**Pull request merged**